// @hocuspocus/extension-database/dist/hocuspocus-database.cjs
Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });
//#region \0rolldown/runtime.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __copyProps = (to, from, except, desc) => {
	if (from && typeof from === "object" || typeof from === "function") {
		for (var keys = __getOwnPropNames(from), i = 0, n = keys.length, key; i < n; i++) {
			key = keys[i];
			if (!__hasOwnProp.call(to, key) && key !== except) {
				__defProp(to, key, {
					get: ((k) => from[k]).bind(null, key),
					enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
				});
			}
		}
	}
	return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", {
	value: mod,
	enumerable: true
}) : target, mod));

//#endregion
let yjs = require("yjs");
yjs = __toESM(yjs);

//#region packages/extension-database/src/Database.ts
var Database = class {
	/**
	* Constructor
	*/
	constructor(configuration) {
		this.configuration = {
			fetch: async () => null,
			store: async () => {}
		};
		this.configuration = {
			...this.configuration,
			...configuration
		};
	}
	/**
	* Get stored data from the database.
	*/
	async onLoadDocument(data) {
		const update = await this.configuration.fetch(data);
		if (update) yjs.applyUpdate(data.document, update);
	}
	/**
	* Store new updates in the database.
	*/
	async onStoreDocument(data) {
		await this.configuration.store({
			...data,
			state: Buffer.from(yjs.encodeStateAsUpdate(data.document))
		});
	}
};

//#endregion
exports.Database = Database;
